### ISSUE TITLE
tags that have illegal characters are silently not created resulting in dirty resources

### ISSUE THREAD

**Issue by powellquiring** [IBM-Cloud/terraform-provider-ibm#2364]
Below look for:   tags = local.tags # this is the problem

Since the tag has an illegal character, '/' the resource is created but the tag is silently not created.  That makes this resource always dirty.  It would be better to fail the creation of the resources when the tag creation fails.

```
export TF_VAR_ibmcloud_api_key=YOUR_API_KEY
export TF_VAR_resource_group_name="default"
export TF_VAR_region="us-south"

terraform {
  required_providers {
    ibm = {
      source  = "IBM-Cloud/ibm"
      version = ">= 1.19.0"
    }
  }
  required_version = ">= 0.14.8"
}



variable ibmcloud_api_key {}
variable region {}
variable resource_group_name {}

provider "ibm" {
  ibmcloud_api_key = var.ibmcloud_api_key
  region           = var.region
  generation       = 2
}

data "ibm_resource_group" "group" {
  name = var.resource_group_name
}

locals {
  tags = [
    "dir:/a",
  ]
}

resource "ibm_is_vpc" "vpc" {
  name = "bug-vpc-tags"
  resource_group = data.ibm_resource_group.group.id
  address_prefix_management = "manual"
  tags = local.tags # this is the problem
}
```

**Comment by kavya498** [comment]
fix is available in 1.23.1 release

**Comment by kavya498** [comment]
Closing this issue..
thank you..
